File: Praktikum-2/Tutor/A/RekeningGiro.cpp
#include "RekeningGiro.h"

RekeningGiro::RekeningGiro(double x, double y) : Rekening(x) {
	if (y < 0) {
		sukuBunga = 0;
	} else {
		sukuBunga = y;
	}
}

void RekeningGiro::setSukuBunga(double x) {
	sukuBunga = x;
}

double RekeningGiro::getSukuBunga() const {
	return sukuBunga;
}

double RekeningGiro::hitungBunga() {
	return Rekening::getSaldo()*sukuBunga;
}

File: Praktikum-4/Tutor/C/RekeningGiro.java
class RekeningGiro extends Rekening {
	protected boolean penalti;

	public RekeningGiro(String nama, double saldo) {
		super(nama, saldo, 0.07);
		this.penalti = false;
		if (this.saldo < 500) {
			this.penalti = true;
		}
	}

	public void setor(double x) {
		this.saldo += x - 0.1;
		if (this.saldo < 500) {
			this.penalti = true;
		}
	}

	public void tarik(double x) {
		if (this.saldo > x + 0.1) {
			this.saldo -= x + 0.1;
		}
		if (this.saldo < 500) {
			this.penalti = true;
		}
	}

	public void update() {
		this.saldo += this.getSukuBunga() * this.saldo - this.hitungBiaya();
		if (this.penalti == true) {
			this.saldo -= 10;
		}
		if (this.saldo > 500) {
			this.penalti = false;
		}
		if (this.saldo < 0) {
			this.saldo = 0;
		}
	}
}